lib/go-atscfg/atscfg.go

```diff
@@ -0,0 +1,114 @@
+	DomainName                    string
+	HostName                      string
+	ID                            int
+	IP                            string
+	ParentCacheGroupID            int
+	ParentCacheGroupType          string
+	ProfileID                     ProfileID
+	ProfileName                   string
+	Port                          int
+	SecondaryParentCacheGroupID   int
+	SecondaryParentCacheGroupType string
+	Type                          string
+}
+
+func (s *ServerInfo) IsTopLevelCache() bool {
+	return (s.ParentCacheGroupType == tc.CacheGroupOriginTypeName || s.ParentCacheGroupID == InvalidID) &&
+		(s.SecondaryParentCacheGroupType == tc.CacheGroupOriginTypeName || s.SecondaryParentCacheGroupID == InvalidID)
+}
+
+func HeaderCommentWithTOVersionStr(name string, nameVersionStr string) string {
+	return "# DO NOT EDIT - Generated for " + name + " by " + nameVersionStr + " on " + time.Now().Format(HeaderCommentDateFormat) + "\n"
+}
+
+func GetNameVersionStringFromToolNameAndURL(toolName string, url string) string {
+	return toolName + " (" + url + ")"
+}
+
+func GenericHeaderComment(name string, toolName string, url string) string {
+	return HeaderCommentWithTOVersionStr(name, GetNameVersionStringFromToolNameAndURL(toolName, url))
+}
+
+// GetATSMajorVersionFromATSVersion returns the major version of the given profile's package trafficserver parameter.
+// The atsVersion is typically a Parameter on the Server's Profile, with the configFile "package" name "trafficserver".
+// Returns an error if atsVersion is empty or not a number.
+func GetATSMajorVersionFromATSVersion(atsVersion string) (int, error) {
+	if len(atsVersion) == 0 {
+		return 0, errors.New("ats version missing")
+	}
+	atsMajorVer, err := strconv.Atoi(atsVersion[:1])
+	if err != nil {
+		return 0, errors.New("ats version parameter '" + atsVersion + "' is not a number")
+	}
+	return atsMajorVer, nil
+}
+
+type DeliveryServiceID int
+type ProfileID int
+type ServerID int
+
+// GenericProfileConfig generates a generic profile config text, from the profile's parameters with the given config file name.
+// This does not include a header comment, because a generic config may not use a number sign as a comment.
+// If you need a header comment, it can be added manually via ats.HeaderComment, or automatically with WithProfileDataHdr.
+func GenericProfileConfig(
+	paramData map[string]string, // GetProfileParamData(tx, profileID, fileName)
+	separator string,
+) string {
+	text := ""
+	for name, val := range paramData {
+		name = trimParamUnderscoreNumSuffix(name)
+		text += name + separator + val + "\n"
+	}
+	return text
+}
+
+// trimParamUnderscoreNumSuffix removes any trailing "__[0-9]+" and returns the trimmed string.
+func trimParamUnderscoreNumSuffix(paramName string) string {
+	underscorePos := strings.LastIndex(paramName, `__`)
+	if underscorePos == -1 {
+		return paramName
+	}
+	if _, err := strconv.ParseFloat(paramName[underscorePos+2:], 64); err != nil {
```

Comment:
I think changing it would be better. If someone _did_ start using 1.1, they might be confused as to why 1.1.4 didn't work the way they expected. And why `name__03E7` works but `name__03D7` doesn't work would be even more confusing. This feature is already hard enough to understand, I think this is a place to be very conservative.